contrib/podex/podex.go

```diff
@@ -28,26 +28,26 @@ limitations under the License.
 package main
 
 import (
-	"bytes"
 	"encoding/json"
 	"flag"
 	"fmt"
-	"io/ioutil"
 	"log"
 	"net/http"
 	"os"
 	"strconv"
 	"strings"
 
-	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/v1beta1"
 	"github.com/ghodss/yaml"
+	goyaml "gopkg.in/v2/yaml"
 )
 
-const usage = "usage: podex [-json|-yaml] [-id PODNAME] IMAGES"
+const usage = "usage: podex [-yaml|-json] [-pod|-container] [-id PODNAME] IMAGES..."
 
+var generateYAML = flag.Bool("yaml", true, "generate yaml manifest (default)")
 var generateJSON = flag.Bool("json", false, "generate json manifest")
-var generateYAML = flag.Bool("yaml", false, "generate yaml manifest")
-var podName = flag.String("id", "", "set pod name")
+var generatePod = flag.Bool("pod", true, "generate pod manifest (default)")
+var generateContainer = flag.Bool("container", false, "generate container manifest")
```

Comment:
suggest a "type" flag which is a string instead of a bunch of booleans. You don't actually want 2^N control paths here.